### PR TITLE
Introduce smart types for pub key hashes and wallet IDs

### PR DESCRIPTION
**Pull request by jhbertra**
Making a PR for this because I want to make sure the ideas in here are actually good ones.

### CODE REVIEW

**Review by bwbush — APPROVED** [review]
I think this does a good job of better organizing the wallet types. In the longer term, I think it could be feasible to (nearly) eliminate the use of public key hashes, which will eventually need to be addresses.

File: marlowe-dashboard-client/src/Capability/Marlowe.purs [hunk 1]
```diff
@@ -295,22 +296,10 @@ instance manageMarloweAppM :: ManageMarlowe AppM where
...
   subscribeToPlutusApp = toBack >>> Left >>> Subscribe >>> sendWsMessage
   subscribeToWallet =
-    toBack
-      >>> un Back.Wallet
-      >>> _.getWalletId
-      >>> (\x -> Back.PubKeyHash { getPubKeyHash: x })
-      >>> Right
-      >>> Subscribe
-      >>> sendWsMessage
+    sendWsMessage <<< Subscribe <<< Right <<< WI.toPubKeyHash
```
Comment (hrajchert) [inline marlowe-dashboard-client/src/Capability/Marlowe.purs:299]: This uses the toPubKeyHash which is incorrect, but I noticed that `sendWsMessage` is using `CombinedWSStreamToServer` which is defined in the PAB and exported as 

```purescript
data CombinedWSStreamToServer
  = Subscribe (Either ContractInstanceId PubKeyHash)
  | Unsubscribe (Either ContractInstanceId PubKeyHash)
```
I think that type is wrong and the PubKeyHash should be changed to WalletId. But that is out of the scope of this PR.

File: marlowe-dashboard-server/src/Marlowe/Run/Wallet/V1/Types.hs [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import qualified Cardano.Wallet.Primitive.Types as Cardano
+import qualified Data.Aeson as Aeson
+import Data.Aeson.Types (parseFail)
+import qualified Data.Text as T
+import Data.Text.Class (FromText (..), TextDecodingError (..), ToText (..))
+import Ledger.Address (PaymentPubKeyHash)
+import Servant (FromHttpApiData (..))
+
+newtype WalletId = WalletId { unWalletId :: Cardano.WalletId }
```
Comment (hrajchert) [inline marlowe-dashboard-server/src/Marlowe/Run/Wallet/V1/Types.hs:16]: Why do we define WalletId here? By doing this aren't we making a difference between the WalletId needed by the PAB and the one used by the dashboard-server?

**Review by hrajchert — APPROVED** [review]
LGTM... minor notes

File: marlowe-dashboard-server/src/Marlowe/Run/Wallet/V1/Types.hs (hunk 2 above)
Comment (jhbertra) [reply marlowe-dashboard-server/src/Marlowe/Run/Wallet/V1/Types.hs:16]: For the instances (the wrapped type does not provide them)

**Pull request merged**